sqlserver/datadog_checks/sqlserver/stored_procedures.py

```diff
@@ -0,0 +1,186 @@
+# (C) Datadog, Inc. 2023-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+import time
+
+from datadog_checks.base import is_affirmative
+from datadog_checks.base.utils.db.statement_metrics import StatementMetrics
+from datadog_checks.base.utils.db.utils import (
+    DBMAsyncJob,
+    default_json_event_encoding,
+)
+from datadog_checks.base.utils.serialization import json
+from datadog_checks.base.utils.tracking import tracked_method
+
+try:
+    import datadog_agent
+except ImportError:
+    from ..stubs import datadog_agent
+
+from datadog_checks.sqlserver.const import STATIC_INFO_ENGINE_EDITION, STATIC_INFO_VERSION
+
+DEFAULT_COLLECTION_INTERVAL = 60
+
+SQL_SERVER_PROCEDURE_METRICS_COLUMNS = [
+    "execution_count",
+    "total_worker_time",
+    "total_physical_reads",
+    "total_logical_writes",
+    "total_logical_reads",
+    "total_elapsed_time",
+    "total_spills",
+]
+
+# TODO: we are doing TOP here, we should use some ORDER BY
```

Comment:
we don't use `ORDER BY` in most places bc of the performance impact it can have on the query. I would imagine that would get expensive for dbs with lots of procedures, but we can test it in the integration env

Reply:
That's what I was thinking as well but wasn't sure.